modules/howtos/pages/provisioning-cluster-resources.adoc

```diff
@@ -141,21 +156,74 @@ include::example$collection-manager.go[tag=scopeAdmin, indent=0]
+It can be instantiated through the `Cluster.QueryIndexes()` method.
+
+[source,golang]
+----
+include::example$query-index-manager.go[tag=creating-index-mgr,indent=0]
+----
+
+include::{version-server}@sdk:shared:partial$flush-info-pars.adoc[tag=query-index-manager-intro]
+
+The example below shows how to create a simple primary index, restricted to a named scope and collection, by calling the `CreatePrimaryIndex()` function.
+Note that you cannot provide a named scope or collection separately, both must be set for the `QueryIndexManager` to create an index on the relevant keyspace path. 
+
+.Creating a primary index
+
+[source,golang]
+----
+include::example$query-index-manager.go[tag=primary,indent=0]
+----
+
+When a primary index name is not specified, the SDK will create the index as `#primary` by default.
+However, if you wish to provide a custom name, you can simply set a `CustomName` property in the `CreatePrimaryQueryIndexOptions` struct.
+
+You may have noticed that the example also sets the `IgnoreIfExists` boolean flag. 
+When set to `true`, this optional argument ensures that an error is not thrown if an index under the same name already exists.
 
-* QueryIndexManager -- https://pkg.go.dev/github.com/couchbase/gocb/v2?tab=doc#Cluster.QueryIndexes[`Cluster.QueryIndexes()`];
-* AnalyticsIndexManager -- https://pkg.go.dev/github.com/couchbase/gocb/v2?tab=doc#Cluster.AnalyticsIndexes[`Cluster.AnalyticsIndexes()`];
-* SearchIndexManager -- https://pkg.go.dev/github.com/couchbase/gocb/v2?tab=doc#Cluster.SearchIndexes[`Cluster.SearchIndexes()`];
-* ViewIndexManager -- https://pkg.go.dev/github.com/couchbase/gocb/v2?tab=doc#Bucket.ViewIndexes[`Bucket.ViewIndexes()`].
+Creating a _secondary_ index follows a similar approach, with some minor differences:
 
-// * Query
+.Creating a secondary index
+
+[source,golang]
+----
+include::example$query-index-manager.go[tag=secondary,indent=0]
+----
+
+The `CreateIndex()` function requires an index name to be provided, along with the fields to create the index on.
+Like the _primary_ index, you can restrict a _secondary_ index to a named scope and collection by passing some options.
+
+Indexes can easily take a long time to build if they contain a lot of documents.
+In these situations, it is more ideal to build indexes in the background.
+To achieve this we can use the `Deferred` boolean option, and set it to `true`.
+
+.Deferring index creation
+
+[source,golang]
+----
+include::example$query-index-manager.go[tag=defer-indexes,indent=0]
+----
+
+To delete a query index you can use the `DropIndex()` or `DropPrimaryIndex()` functions. 
+Which one you use depends on the type of query index you wish to drop from the database.
+
+.Deleting an index
+
+[source,golang]
+----
+include::example$query-index-manager.go[tag=drop-primary-or-secondary-index,indent=0]
+----
 
 // * Search - note & link to FTS page & API?
 
-== View Management
+== Views Management
```

Comment:
Oh, why is this plural btw? We don't say "Indexes management" for example.

Oh, is it because of ambiguity of "View" (verb/noun)? Hmm. OK, still, seems a bit inconsistent, only just noticed this time.

Reply:
Views is short for Map/Reduce Views.